### DISCUSSION TITLE
Where to find username

### DISCUSSION THREAD

**Discussion by guillezcurra** [tomtau/superbasic-im#1] [Q&A] [answered]
I have deployed this project in my local server and now I have access to the login page from my local LAN.
I have done the node index.js init and add my phone number and password.
When I access the login page it asks for **username** and password. I have tried to put my phone number as the username but the answer is always 403 forbidden.
Am I missing something? How do I know the username?

Thank you.

**Comment by tomtau** [comment, marked as answer]
username should be the phone number you entered (you can check the user.json file).

**Reply by guillezcurra** [reply]
Thanks for your answer, I try to use my phone as username and password and the answer is always:
{
    "statusCode": 403,
    "error": "Forbidden",
    "message": "Forbidden"
}
I check the file user.json and verified my phone number is correct.
I cannot find any log explaining why the error, the only thing I cound find checking "pm2 logs" is this trace:
0|index    |     TypeError: Cannot read properties of null (reading 'evaluate')
0|index    |     at Client.getState (/var/www/superbasic-im/node_modules/whatsapp-web.js/src/Client.js:1119:35)
0|index    |     at /var/www/superbasic-im/dist/index.js:89:53
0|index    |     at Generator.next (<anonymous>)
0|index    |     at /var/www/superbasic-im/dist/index.js:31:71
0|index    |     at new Promise (<anonymous>)
0|index    |     at __awaiter (/var/www/superbasic-im/dist/index.js:27:12)
0|index    |     at /var/www/superbasic-im/dist/index.js:85:57
0|index    |     at exports.Manager.execute (/var/www/superbasic-im/node_modules/@hapi/hapi/lib/toolkit.js:57:29)
0|index    |     at Request._invoke (/var/www/superbasic-im/node_modules/@hapi/hapi/lib/request.js:397:55)
0|index    |     at Request._onRequest (/var/www/superbasic-im/node_modules/@hapi/hapi/lib/request.js:289:41)
0|index    |     at Request._execute (/var/www/superbasic-im/node_modules/@hapi/hapi/lib/request.js:271:24)
0|index    |     at Server.<anonymous> (/var/www/superbasic-im/node_modules/@hapi/hapi/lib/core.js:543:21)
0|index    |     at Server.emit (node:events:513:28)
0|index    |     at Server.<anonymous> (/usr/local/lib/node_modules/pm2/node_modules/@pm2/io/build/main/metrics/httpMetrics.js:166:33)
0|index    |     at parserOnIncoming (node:_http_server:1086:12)
0|index    |     at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)

It appears everytime I load my SERVER_IP:4000/login page.
Do you know if this can have anything to do?

Sorry to bother you

**Reply by tomtau** [reply]
Possibly, it seems it failed to initialise whatsapp-web.js's Client. Do you see "Client is ready!" log message in there anywhere?

**Reply by guillezcurra** [reply]
I can't find that message in log, but I have found this message when executing "sudo pm2 restart index":

Error: Failed to launch the browser process!
[0905/134808.125025:ERROR:zygote_host_impl_linux.cc(90)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.

TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md
at onClose (/var/www/superbasic-im/node_modules/puppeteer-core/lib/cjs/puppeteer/node/BrowserRunner.js:299:20)
...

I am looking what may I have wrong configured.

**Reply by tomtau** [reply]
> Running as root without --no-sandbox is not supported.

I guess you're executing it as a root (Linux) user? Can you run it under a less privileged user?

**Reply by guillezcurra** [reply]
Using a non root user it seems that the app has started OK but when i try to log in I still have this issue:
<img width="367" alt="image" src="https://github.com/user-attachments/assets/69e17113-9d72-4692-9ce8-18404c6d13fc">

In the pm2 logs I see the message:
0|index    | QR code RECEIVED
0|index    | QR code RECEIVED
0|index    | QR code RECEIVED
0|index    | QR code RECEIVED
0|index    | QR code RECEIVED
Being logged every tyme.

Thank you for your help.

**Reply by guillezcurra** [reply]
Hello, I could make it work, this is amazing, thank you!

**Reply by tomtau** [reply]
@guillezcurra that's great! what was the final issue / how did you resolve it?

**Reply by guillezcurra** [reply]
When I put the application on the internet it started to work.

**Reply by dariolob** [reply]
I think this is not 100% related with exposing the application on the internet: in my test environment I was running it with no restrictions on outbound traffic, but it still wasn't working, with the same issues described by @guillezcurra 
Enabling HTTPS protocol (using ngrok as reverse proxy, in my case) did the trick, and after login I was able to see the QR code. Once past this stage, the application can be used again HTTP-only, so the culprit here is very likely whatsapp-web.js.

**Reply by petarmarinov37** [reply]
Hello, I was getting the same error at times, but am now getting this:

`{"statusCode":500,"error":"Internal Server Error","message":"An internal server error occurred"} `

I am using nginx via docker. I am not sure how to “put the application on the internet” or “enable HTTPS protocol” as I thought I had already done all that by setting up nginx… I should mention, I am only using docker because I am under the impression that it is required in order to use nginx with npm, so please correct me if there’s a simpler way. I have already tried editing the nginx.conf file within the docker container to tell it to listen on port 80 and proxy to 4000, and nothing changed, except now even the nginx manager page at 81 no longer works.
